plugins/org.ruyisdk.intro/src/org/ruyisdk/intro/CustomIntroPart.java

```diff
@@ -68,82 +61,65 @@ public void createPartControl(Composite parent) {
-                        System.out.println("DEBUG: CustomIntroPart: Successfully resolved resource '" + resourcePath + "' to: " + fileUrl.toExternalForm());
+                        URL fileUrl = FileLocator.toFileURL(bundleUrl); 
                         if (resourcePath.equals("html/welcome.html")) {
-                            resolvedWelcomePageURL = fileUrl; // 保存 welcome.html 的 file: URL
+                            resolvedWelcomePageURL = fileUrl; 
                         }
                     } catch (IOException e_resolve) {
-                        System.err.println("DEBUG: CustomIntroPart: Failed to resolve resource '" + resourcePath + "' using FileLocator.toFileURL(). Error: " + e_resolve.getMessage());
-                        // 即使某个资源解析失败，也继续尝试其他资源，但主 HTML 必须成功
                         if (resourcePath.equals("html/welcome.html")) {
-                             System.err.println("DEBUG: CustomIntroPart: CRITICAL - Failed to resolve welcome.html. Cannot proceed.");
                              browser.setText("<html><body><h1>Error: Could not resolve welcome.html to a file URL.</h1><p>" + e_resolve.getMessage() + "</p></body></html>");
                              return;
                         }
                     }
                 } else {
-                    System.err.println("DEBUG: CustomIntroPart: Resource NOT FOUND in bundle: " + resourcePath);
                      if (resourcePath.equals("html/welcome.html")) {
-                         System.err.println("DEBUG: CustomIntroPart: CRITICAL - welcome.html not found in bundle. Cannot proceed.");
                          browser.setText("<html><body><h1>Error: welcome.html not found in bundle.</h1></body></html>");
                          return;
                      }
                 }
             }
 
             if (resolvedWelcomePageURL != null) {
-                System.out.println("INFO: CustomIntroPart: All (attempted) resources touched. Setting browser URL to resolved welcome.html: " + resolvedWelcomePageURL.toExternalForm());
-                browser.setUrl(resolvedWelcomePageURL.toExternalForm()); // 使用指向解压后 welcome.html 的 file: URL
+                browser.setUrl(resolvedWelcomePageURL.toExternalForm()); 
             } else {
-                System.err.println("DEBUG: CustomIntroPart: CRITICAL - resolvedWelcomePageURL is null after attempting to touch all resources. This should not happen if welcome.html was found.");
                 browser.setText("<html><body><h1>Error: Welcome page URL could not be determined after resource processing.</h1></body></html>");
             }
 
-        } catch (Exception e) { // 捕获其他意外错误
-            System.err.println("DEBUG: CustomIntroPart: General error occurred while loading welcome page.");
-            e.printStackTrace(); 
+        } catch (Exception e) { 
+
             browser.setText("<html><body><h1>Error loading welcome page.</h1><p>" + e.getMessage() + "</p></body></html>");
```

Comment:
I think this place is unsafe. If the error message contains special characters such as "<", it will produce invalid HTML. 

I've noticed several other instances of such unsafe usage - please consider creating a helper method to escape special characters.